### PR TITLE
Fix 3803. Added support for oneOf and anyOf for Example Value

### LINKED ISSUE #3803: Multiple responses using oneOf attribute do not appear in UI
| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  Bug
| Which Swagger/OpenAPI version?  |
| Which Swagger-UI version?       |
| How did you install Swagger-UI? | 
| Which browser & version?        |
| Which operating system?         | 

### Demonstration API definition

```yaml
responses:
  '200':
    description: OK
    content:
      application/json:
        schema:
          oneOf:
            - $ref: '#/components/schemas/AuthorisationResponse'
            - $ref: '#/components/schemas/ForbiddenResponse'
```

### Configuration (browser query string, constructor, config.yaml)

```js
constructorConfig contains just the default values.
```
### Expected Behavior
Since I have two schemas passed in as a $ref to the 200 response I would expect to see two schemas in the Swagger UI application.  

### Current Behavior
Instead I see only the reference to the 200 response and its description with no example responses displayed as JSOn in the black boxes.

### PR DESCRIPTION
**Pull request by PavelStefanov**
### Motivation and Context
Fixes #3803 

### Description
I added support for oneOf and anyOf attributes for example value tab. 

If we have this scheme
``` yaml
/users:
    post:
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                oneOf:
                  - $ref: '#/components/schemas/Jessica'
                  - $ref: '#/components/schemas/Ron'
                  - $ref: '#/components/schemas/dddddd'
                  - type: object
                    properties:
                      id:
                        type: integer
                        format: int64
                      name:
                        type: string
```

So it will look like this

image

Each subscheme will be rendered with name and in separate highlight block.
I'll be glad to your ideas and comments!

### How Has This Been Tested?
I tested different cases of oneOf and anyOf, but I'm not sure I was able to cover all cases.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/components/model-example.js [hunk 1]
```diff
@@ -72,7 +72,6 @@ describe("<ModelExample/>", function(){
       expect(modelTab.hasClass("active")).toEqual(false)
 
       expect(wrapper.find("div > div").length).toEqual(1)
-      expect(wrapper.find("div > div").text()).toEqual(props.example)
```
Comment (heldersepu) [inline test/components/model-example.js]: No UnitTests?

**Comment by shockey** [comment]
@webron, could use your take on the visual changes

**Comment by alvarolb** [comment]
Nice feature!

It would be great to have this also on requests. Not sure if it is so complex, but It will be great if instead of showing all possible cases, just place a select control to select one of the possible cases. 

Best.

**Comment by webron** [comment]
@PavelStefanov Thanks for the PR and sorry for the delay in reviewing it. I have a few comments, but they are all also related to dealing with the bigger issue at hand. I'll take the discussion back to #3803 to get more interaction from the community at large.

**Comment by codedmonkey** [comment]
Thank you for trying to create a solution for the problem. There still is an issue with the code when you use multiple content types, as the `activeTab` value doesn't get updated when switching between those.

**Comment by shockey** [comment]
closing due to inactivity!

**Pull request closed without merging**